common/src/androidTest/java/com/microsoft/identity/common/internal/platform/DevicePoPManagerTests.java

```diff
@@ -289,4 +302,60 @@ public void testMintSignedAccessTokenContainsRequisiteClaims()
         Assert.assertEquals(nonce, jwtClaimsSet.getClaim("nonce"));
         Assert.assertNotNull(jwtClaimsSet.getClaim("cnf"));
     }
+
+    @Test
+    public void testAsymmetricKeyNullWhenUninitialized() throws ClientException {
+        final Date createdDate = mDevicePopManager.getAsymmetricKeyCreationDate();
+        Assert.assertNull(createdDate);
+    }
+
+    @Test
+    public void testAsymmetricKeyHasCreationDate() throws ClientException {
+        final Date createdDate = mDevicePopManager.getAsymmetricKeyCreationDate();
+        Assert.assertNull(createdDate);
+
+        // Generate it
+        mDevicePopManager.generateAsymmetricKey(mContext);
+
+        // Assert the Date exists
+        Assert.assertNotNull(mDevicePopManager.getAsymmetricKeyCreationDate());
+    }
+
+    @Test
+    public void testAsymmetricKeyHasPublicKeyX509() throws ClientException, NoSuchAlgorithmException, InvalidKeySpecException {
+        // Generate keys
+        mDevicePopManager.generateAsymmetricKey(mContext);
+
+        // Get the public key
+        final String publicKey = mDevicePopManager.getPublicKey(X_509_SubjectPublicKeyInfo_ASN_1);
+
+        // Rehydrate the certificate
+        final byte[] bytes = Base64.decode(publicKey, Base64.DEFAULT);
+        final KeyFactory keyFactory = KeyFactory.getInstance("RSA");
+        final PublicKey pubKeyRestored = keyFactory.generatePublic(new X509EncodedKeySpec(bytes));
+        Assert.assertEquals("X.509", pubKeyRestored.getFormat());
+    }
+
+    @Test
+    public void testAsymmetricKeyHasPublicKeyJwk() throws ClientException {
+        // Generate keys
+        mDevicePopManager.generateAsymmetricKey(mContext);
+
+        // Get the public key
+        final String publicKey = mDevicePopManager.getPublicKey(JWK);
+
+        // Convert it to JSON, parse to verify fields
+        final JsonElement jwkElement = new JsonParser().parse(publicKey);
+
+        // Convert to JsonObject to extract claims
+        final JsonObject jwkObj = jwkElement.getAsJsonObject();
+
+        // We should expect the following claims...
+        // 'kty' - Key Type - Identifies the cryptographic alg used with this key (ex: RSA, EC)
+        // 'e' - Public Exponent - The exponent used on signed/encoded data to decode the orig value
+        // 'n' - Modulus - The product of two prime numbers used to generate the key pair
+        Assert.assertNotNull(jwkObj.get("kty"));
+        Assert.assertNotNull(jwkObj.get("e"));
+        Assert.assertNotNull(jwkObj.get("n"));
```

Comment:
What data type are these? If these are Strings, should we also assert not empty String?

Reply:
d09089bff82959befd903fe5c13f5ed81405e6aa